Add committee bylaws (#360)
ghc-proposals/ghc-proposals: acceptance.rst

How to Accept a Proposal
========================

This document describes the technical process for accepting a proposal,
assuming the proposal has gone through the `proposal process <https://github.com/ghc-proposals/ghc-proposals/#committee-process>`_.

Acceptance checklist
--------------------

> Before accepting, the shepherd should check for the following

Reviewer: Also: before sending the proposal to the committee to review.

In my view, the first job of the shepherd, when handed a proposal, is to help make sure that the checks below pass. This means both ensuring that the conditions are met before sending the proposal over to the committee, and, when they aren't, to help guide the proposal's authors towards meeting said conditions.

Author: Yes, but it's very easy at that stage to focus on the substance of the proposal, instead of these details. And I personally think that's OK. But we should make sure the details are attended to in the end.

Reviewer: I guess what I'm getting at is that we should use this opportunity to set the expectations that we have for the shepherd. I'm guessing that we don't have the same image of the role. When I'm shepherding, I try to suspend my judgement as much as possible and focus on ensuring that the proposal is the best proposal it can be, regardless of whether I'm going to recommend acceptance or rejection.

2. Is the proposal precise? The specification section should fully describe
   the proposal payload. In particular:

   a. If there is new syntax, is it described in BNF notation?

   b. If applicable, are there typing rules? If there are not, could
      typing rules be reasonably extracted from the proposal text?

   c. Are any changes to runtime behavior precisely specified?

   d. Is there a reasonable description of how the new feature might
      be documented in Haddock? Linted by HLint? Treated by other tools?
      (These will not apply to every proposal.)

3. Is the proposal sufficiently illustrated with examples? Examples help
   us understand the details that will be needed during implementation.

Acceptance steps
----------------

A committee member accepts a proposal by following this sequence of
steps:

1. Add a new commit on top of the PR branch that:

   a. Changes the filename of the proposal to correspond to the PR number.

   b. Updates any metadata fields that may have changed in the template on ``master`` since
      the PR branch split off.

   c. Fills in these metadata fields as appropriate, including changing "is discussed"
      to "was discussed".

> 2. Merge the PR branch into master, and push.

Reviewer 2: I usually merge master into the PR, then update the metadata, then push to master. But we probably don’t care too much about the various ways of achieving the same result, so no change needed.

3. Update the PR description to start
   with the text "The proposal has been accepted; the following discussion is mostly of historic interest."

>    where the word "proposal" links to the final rendered version pushed above.

Reviewer 2: ```suggestion
   where the word "proposal" links to the final rendered version, as found on https://github.com/ghc-proposals/ghc-proposals/tree/master/proposals
```
